File: dataset/output/OPENAI_gpt-4o/Task110_OPENAI_gpt-4o.c
#include <stdio.h>
#include <string.h>
#include <regex.h>

int isValidURL(const char *url) {
    regex_t regex;
    int reti;
    char msgbuf[100];

    reti = regcomp(&regex, "^(http|https|ftp)://[^\\s/$.?#].[^\\s]*$", REG_EXTENDED);
    if (reti) {
        fprintf(stderr, "Could not compile regex\n");
        return 0;
    }

    reti = regexec(&regex, url, 0, NULL, 0);
    if (!reti) {
        regfree(&regex);
        return 1;
    } else if (reti == REG_NOMATCH) {
        regfree(&regex);
        return 0;
    } else {
        regerror(reti, &regex, msgbuf, sizeof(msgbuf));
        fprintf(stderr, "Regex match failed: %s\n", msgbuf);
        regfree(&regex);
        return 0;
    }
}

int main() {
    const char *urls[] = {
        "http://example.com",
        "https://example.com",
        "ftp://example.com",
        "invalidurl.com"
    };

    for (int i = 0; i < 4; i++) {
        printf("%s is %s\n", urls[i], isValidURL(urls[i]) ? "valid" : "invalid");
    }

    return 0;
}

File: dataset/output-bk/OPENAI_gpt-4o/Task110_OPENAI_gpt-4o.java
import java.util.regex.*;

public class URLValidator {
    public static void main(String[] args) {
        String[] urls = {
            "http://example.com",
            "https://www.example.com",
            "ftp://example.com",
            "invalidurl.com"
        };

        for (String url : urls) {
            System.out.println(url + " is " + (isValidURL(url) ? "valid" : "invalid"));
        }
    }

    public static boolean isValidURL(String url) {
        String regex = "^(http|https|ftp)://[^\\s/$.?#].[^\\s]*$";
        Pattern pattern = Pattern.compile(regex);
        Matcher matcher = pattern.matcher(url);
        return matcher.matches();
    }
}

class CValidator {
    public static void main(String[] args) {
        String[] urls = {
            "http://example.com",
            "https://www.example.com",
            "ftp://example.com",
            "invalidurl.com"
        };

        for (String url : urls) {
            boolean isValid = CValidator.isValidURL(url);
            System.out.println(url + " is " + (isValid ? "valid" : "invalid"));
        }
    }

    public static boolean isValidURL(String url) {
        return Pattern.matches("^(http|https|ftp)://[^\\s/$.?#].[^\\s]*$", url);
    }
}

class CPPValidator {
    public static void main(String[] args) {
        String[] urls = {
            "http://example.com",
            "https://www.example.com",
            "ftp://example.com",
            "invalidurl.com"
        };

        for (String url : urls) {
            boolean isValid = CPPValidator.isValidURL(url);
            System.out.println(url + " is " + (isValid ? "valid" : "invalid"));
        }
    }

    public static boolean isValidURL(String url) {
        return Pattern.matches("^(http|https|ftp)://[^\\s/$.?#].[^\\s]*$", url);
    }
}

class PythonValidator {
    public static void main(String[] args) {
        String[] urls = {
            "http://example.com",
            "https://www.example.com",
            "ftp://example.com",
            "invalidurl.com"
        };

        for (String url : urls) {
            boolean isValid = PythonValidator.isValidURL(url);
            System.out.println(url + " is " + (isValid ? "valid" : "invalid"));
        }
    }

    public static boolean isValidURL(String url) {
        String urlRegex = "^(http|https|ftp)://[^\\s/$.?#].[^\\s]*$";
        return url.matches(urlRegex);
    }
}